Source/FieldSolver/SpectralSolver/SpectralFieldData.cpp

```diff
@@ -66,8 +66,15 @@ SpectralFieldIndex::SpectralFieldIndex (const bool update_with_rho,
         {
             Jx_new = c++; Jy_new = c++; Jz_new = c++;
 
-            // TODO Allocate F and G only when needed
-            F = c++; G = c++;
+            if (dive_cleaning)
+            {
+                F = c++;
```

Comment:
Just double checking for all the values here: you want to assign `c` and then increase `c` afterwards, right?

Reply:
Yes, that's right. First assign, and then increase.